packages/adblocker-webextension/test/index.test.ts

```diff
@@ -125,3 +131,209 @@ describe('#fromWebRequestDetails', () => {
+
+      public status: WebRequest.StreamFilterStatus = 'uninitialized';
+      public error: string = '';
+
+      private isResumable() {
+        return (
+          this.status !== 'closed' && this.status !== 'failed' && this.status !== 'disconnected'
+        );
+      }
+
+      public create(_requestId: number, _addonId: string): void {
+        throw new Error('StreamFilter.create() is not implemented!');
+      }
+
+      public suspend(): void {
+        this.status = 'suspended';
+      }
+
+      public resume(): void {
+        if (this.isResumable() === false) {
+          this.error = 'This instance is not resumable!';
+          this.onerror({
+            data: Uint8Array.from([]),
+          });
+          return;
+        }
+        this.status = 'transferringdata';
+      }
+
+      public close(): void {
+        this.status = 'closed';
+      }
+
+      public disconnect(): void {
+        this.status = 'disconnected';
+        this.ondata = this._ondata;
+        this.onstop = this._noop;
+      }
+
+      public write(data: ArrayBuffer | Uint8Array): void {
+        const next = new Uint8Array(this._pulled.byteLength + data.byteLength);
+        next.set(this._pulled);
+        next.set(new Uint8Array(data), this._pulled.byteLength);
+        this._pulled = next;
+      }
+
+      public _ondata(event: WebRequest.StreamFilterEventData) {
+        this.write(event.data);
+      }
+
+      public _noop() {}
+
+      public _push(data: ArrayBuffer | Uint8Array, isLast: boolean = false) {
+        if (this.status !== 'transferringdata' && this.status !== 'uninitialized') {
+          this.error = 'Further data transfer cannot be done since the stream is already closed!';
+          this.onerror({
+            data: Uint8Array.from([]),
+          });
+          return;
+        }
+
+        if (this._pushed === 0) {
+          this.status = 'transferringdata';
+        }
+
+        this.ondata({ data });
+        if (isLast === true) {
+          this.onstop({ data: Uint8Array.from([]) });
+          this.status = 'finishedtransferringdata';
+        }
+      }
+
+      public _reset() {
+        this._pulled = new Uint8Array();
+        this._pushed = 0;
+        this.status = 'uninitialized';
+        this.error = '';
+        this.ondata = this._ondata;
+        this.onstart = this._noop;
+        this.onstop = this._ondata;
+        this.onerror = this._noop;
+      }
+    }
+
+    it('stops processing more than MAXIMUM_RESPONSE_BUFFER_SIZE', () => {
+      const streamFilter = new StreamFilter();
+      const details: OnHeadersReceivedDetailsType = {
+        responseHeaders: [
+          {
+            name: 'content-type',
+            value: 'text/text',
+          },
+        ],
+        url: 'https://foo.com/script.js',
+        type: 'main_frame',
+        originUrl: 'https://foo.com/',
+        tabId: 0,
+        requestId: 'req-00',
+      };
+      const request = fromWebRequestDetails(details);
```

Comment:
can be merged together into 
```js
const request = fromWebRequestDetails({
  responseHeaders: [
    {
      name: 'content-type',
      value: 'text/text',
    },
  ],
  url: 'https://foo.com/script.js',
  type: 'main_frame',
  originUrl: 'https://foo.com/',
  tabId: 0,
  requestId: 'req-00',
});
```

Reply:
Updated in [`9991a4a` (#14)](https://github.com/seia-soto/adblocker/pull/14/commits/9991a4a7979f149105991bd4b199be1250a86d41)